lib/Unicode/Char/Numeric.hs

```diff
@@ -21,50 +25,70 @@ module Unicode.Char.Numeric
     ) where
 
 import Data.Char (digitToInt, intToDigit, isDigit, isHexDigit, isOctDigit)
+import Data.Maybe (isJust)
 import Data.Ratio (numerator, denominator)
-import Unicode.Char.General (GeneralCategory(..), generalCategory)
+import qualified Unicode.Char.Numeric.Compat as Compat
 import qualified Unicode.Internal.Char.DerivedNumericValues as V
 
 -- | Selects Unicode numeric characters, including digits from various
 -- scripts, Roman numerals, et cetera.
 --
 -- This function returns 'True' if its argument has one of the
--- following 'GeneralCategory's, or 'False' otherwise:
+-- following 'Unicode.Char.General.GeneralCategory's, or 'False' otherwise:
 --
--- * 'DecimalNumber'
--- * 'LetterNumber'
--- * 'OtherNumber'
+-- * 'Unicode.Char.General.DecimalNumber'
+-- * 'Unicode.Char.General.LetterNumber'
+-- * 'Unicode.Char.General.OtherNumber'
 --
 -- __Note:__ a character may have a numeric value (see 'numericValue') but return
--- 'False', because 'isNumber' only tests 'GeneralCategory':
--- some CJK characters are 'OtherLetter' and do have a numeric value.
+-- 'False', because 'isNumber' only tests 'Unicode.Char.General.GeneralCategory':
+-- some CJK characters are 'Unicode.Char.General.OtherLetter' and do have a
+-- numeric value. Use 'hasNumericValue' to cover those cases as well.
 --
 -- prop> isNumber c == Data.Char.isNumber c
 --
 -- @since 0.3.0
+{-# DEPRECATED isNumber "Use Unicode.Char.Numeric.Compat.isNumber instead. This function will be a synonym for hasNumericValue in a future release. See Unicode.Char.Numeric.Compat for behavior compatible with base:Data.Char." #-}
+{-# INLINE isNumber #-}
 isNumber :: Char -> Bool
-isNumber c = case generalCategory c of
-    DecimalNumber -> True
-    LetterNumber  -> True
-    OtherNumber   -> True
-    _             -> False
+isNumber = Compat.isNumber
+
+-- [TODO] @since
+-- | Selects Unicode character with a numeric value.
+--
+-- __Note:__ a character may have a numeric value but return 'False' with
+-- the predicate 'Unicode.Char.Numeric.Compat.isNumber', because
+-- 'Unicode.Char.Numeric.Compat.isNumber' only tests
+-- 'Unicode.Char.General.GeneralCategory': some CJK characters are
+-- 'Unicode.Char.General.OtherLetter' and do have a numeric value.
+--
+-- prop> hasNumericValue c == isJust (numericValue c)
+{-# INLINE hasNumericValue #-}
+hasNumericValue :: Char -> Bool
+hasNumericValue = isJust . V.numericValue
```

Comment:
We can call it `isNumeric`.

Reply:
@wismill what is your opinion on renaming this to `isNumeric`? That's a shorter name and in line with other predicates.

Reply:
I agree. Done.